internal/dataplane/kong_client.go

```diff
@@ -312,12 +317,18 @@ func (c *KongClient) Update(ctx context.Context) error {
 
 	// parse the Kubernetes objects from the storer into Kong configuration
 	kongstate := p.Build()
-	// todo: does it still make sense to report TranslationCount when Build no longer returns an error?
-	// https://github.com/Kong/kubernetes-ingress-controller/issues/1892
-	c.prometheusMetrics.TranslationCount.With(prometheus.Labels{
-		metrics.SuccessKey: metrics.SuccessTrue,
-	}).Inc()
-	c.logger.Debug("successfully built data-plane configuration")
+	if errors := p.GetParsingErrors(); errors != nil {
+		c.createParsingErrorsEvents(errors)
+		c.prometheusMetrics.TranslationCount.With(prometheus.Labels{
+			metrics.SuccessKey: metrics.SuccessFalse,
+		}).Inc()
+		c.logger.Debugf("%d translation errors occurred when building data-plane configuration", len(errors))
```

Comment:
In case any translation error is detected, we consider that a failed translation thus propagating the metric with `SuccessFalse` value. IMO it makes sense as that will allow setting up alerting easily.